### ISSUE TITLE
[Telegram] "Join to chat" links are opened in external browser instead if processing internally

### ISSUE THREAD

**Issue by MurzNN** [ferdium/ferdium-app#1027]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that Ferdium Recipes adheres to.
- [X] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22recipe+%3Abookmark_tabs%3A%22+label%3A%22bug+%3Abeetle%3A%22+) for a bug report that matches the one I want to file, without success.

### Ferdium Version

6.2.4

### Recipe Name

Telegram

### Recipe Version

3.3.5

### Steps to reproduce

1. Receive a "join" link like this: https://t.me/+o0u9GKz1wMk1Y2M0 in Telegram as an incoming message.
2. Click on it.

### Expected Behavior

You should be redirected to the target chat inside Ferdium with ability to join the chat.

### Actual Behavior

The join link is opened in an external browser instead.

**Comment by MartinX3** [comment]
Same for links of rose bot notes.

If I activate "open links within ferdium", I get the error about an unsupported location.
![image](https://user-images.githubusercontent.com/4518092/227623964-451dc92c-5877-43ee-9641-8797b9fc2f05.png)

**Comment by vraravam** [comment]
@MurzNN - could you please try with the latest version and also use the "open within Ferdium" option?

**Comment by vraravam** [comment]
@MartinX3 - please provide translation in English - only then will we be able to help further

**Comment by MurzNN** [comment]
Checked with the latest version - the issue is here. 

With the "open within Ferdium" option - it opens the "join" page inside Ferdium, but the "Join group" button doesn't work:
![image](https://github.com/ferdium/ferdium-app/assets/336662/fbb4ac3b-d3a3-42b4-a8b2-e43b69643736)

- on click it tries to open the telegram desktop app.

**Comment by marie-bnl** [comment]
As you mentioned, the error with `tg://` is because the “Join group” button tries to open the Telegram application. You need to click “Open in web” instead.

Edit: Actually it happens even if you don’t click the button – because Telegram Web doesn’t wait for you clicking it and tries to open Telegram by itself, and I think there’s not much we can do about it in Ferdium. I think you should just dismiss the error and click “Open in web”.

**Comment by MurzNN** [comment]
> As you mentioned, the error with tg:// is because the “Join group” button tries to open the Telegram application. You need to click “Open in web” instead.

The issue is exactly about the "Open in web" and even about the previous link. So, in a Telegram message, I have a link like https://t.me/some_group_name/1234 that points to a message in the same chat. 

In a full browser using https://web.telegram.org/ - clicking on this link just scrolls the timeline to that message inside the current app.

But in Ferdium - it opens the external web browser instead. That is the main problem.

**Comment by MurzNN** [comment]
Hope https://github.com/ferdium/ferdium-recipes/pull/394 will resolve this issue.

**Comment by MurzNN** [comment]
Tested https://github.com/ferdium/ferdium-recipes/pull/394 (release 3.3.6) and with the message from https://github.com/ferdium/ferdium-app/issues/1027#issuecomment-1645168828 it starts just redirecting me to the https://t.me/some_group_name/1234 page (inside Ferdium) instead of scrolling to the message.

And the same for links like  https://t.me/+o0u9GKz1wMk1Y2M0 (from the initial post) - I get redirected to this page inside Ferdium, instead of "Join" action.

But the good thing is that now with the "Open in web" button on the target page - now we can go back to the Telegram Web interface to the desired target. That is a good improvement, but still not perfect.

So, compared with https://web.telegram.org opened in a web browser, in Ferdium now we have to do two clicks and get the target via the intermediate page, instead of just directly get the result.

**Comment by marie-bnl** [comment]
You’re right, I thought it was just Telegram’s behaviour and happened in the browser as well. However after testing, it turns out it indeed doesn’t. I’ll try to figure out what differs between Ferdium and a browser, that makes Ferdium behave so.

**Comment by xdhtyiauugcbbh-dev** [comment]
> ### Preflight Checklist
> * [x]  I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that Ferdium Recipes adheres to.[x]  I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22recipe+%3Abookmark_tabs%3A%22+label%3A%22bug+%3Abeetle%3A%22+) for a bug report that matches the one I want to file, without success.
> 
> ### Ferdium Version
> 6.2.4
> 
> ### Recipe Name
> Telegram
> 
> ### Recipe Version
> 3.3.5
> 
> ### Last Known working version of the recipe
> _No response_
> 
> ### Steps to reproduce
> 1. Receive a "join" link like this: https://t.me/+o0u9GKz1wMk1Y2M0 in Telegram as an incoming message.
> 2. Click on it.
> 
> ### Expected Behavior
> You should be redirected to the target chat inside Ferdium with ability to join the chat.
> 
> ### Actual Behavior
> The join link is opened in an external browser instead.
> 
> ### Screenshots
> _No response_
> 
> ### Additional Information
> _No response_
